File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/model/CurrentStatus.java
package com.github.seratch.jslack.api.status.v2.model;

import java.util.List;

public class CurrentStatus {
    private String status;
    private String dateCreated;
    private String dateUpdated;
    private List<SlackIssue> activeIncidents;

    public String getStatus() {
        return this.status;
    }

    public String getDateCreated() {
        return this.dateCreated;
    }

    public String getDateUpdated() {
        return this.dateUpdated;
    }

    public List<SlackIssue> getActiveIncidents() {
        return this.activeIncidents;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void setDateCreated(String dateCreated) {
        this.dateCreated = dateCreated;
    }

    public void setDateUpdated(String dateUpdated) {
        this.dateUpdated = dateUpdated;
    }

    public void setActiveIncidents(List<SlackIssue> activeIncidents) {
        this.activeIncidents = activeIncidents;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/model/CurrentStatus.kt
package work.socialhub.kslack.api.status.model

import kotlinx.serialization.Serializable
import kotlin.js.JsExport

/**
 * Represents the current operational status of Slack services.
 *
 * Queried from the public Slack Status API at https://status.slack.com.
 * Contains the overall service status, timestamps, and any active incidents.
 *
 * @see <a href="https://status.slack.com">Slack Status Page</a>
 */
@JsExport
@Serializable
class CurrentStatus {
    var status: String? = null
    var dateCreated: String? = null
    var dateUpdated: String? = null
    var activeIncidents: Array<SlackIssue>? = null
}
